src/pybind/cudamatrix/cu_matrix_pybind.cc

```diff
@@ -37,10 +38,71 @@ void pybind_cu_matrix(py::module& m) {
         .def("SetZero", &PyClass::SetZero)
         .def("Set", &PyClass::Set, py::arg("value"))
         .def("Add", &PyClass::Add, py::arg("value"))
-        .def("Scale", &PyClass::Scale, py::arg("value"));
-    ;
+        .def("Scale", &PyClass::Scale, py::arg("value"))
+        .def("__getitem__",
+             [](const PyClass& m, std::pair<ssize_t, ssize_t> i) {
+               return m(i.first, i.second);
+             })
+#if HAVE_CUDA == 1
+        .def("to_dlpack",
+             [](PyClass* m) {
```

Comment:
Bear in mind that if HAVE_CUDA is not 1, it will actually contain CPU memory.  You might want to eventually handle that case.

Reply:
My intention is that if a user in Python wants to use CuMatrix/CuVector, then he/she means the GPU one. If he wants only CPU, then he/she should use Matrix/Vector.

Reply:
Now it will call KALDI_ERR for `to_dlpack` if kaldi is not compiled with CUDA but the user
calls `to_dlpack` for CuVectorBase/CuMatrixBase.